amr-wind/wind_energy/ABLWallFunction.cpp

```diff
@@ -60,81 +62,194 @@ ABLWallFunction::ABLWallFunction(const CFDSim& sim)
             << std::endl;
     }
 
-    if (pp.contains("surface_temp_flux")) {
-        pp.query("surface_temp_flux", m_mo.surf_temp_flux);
+    // Check for type of surface heating and read in data.
+    // - Specified surface temperature flux mode:
+    if ((pp.contains("surface_temp_flux")) ||
+        (pp.contains("surface_temp_flux_timetable"))) {
+        m_surf_temp_flux = true;
+        m_surf_temp = false;
+        m_nearsurf_temp = false;
+
+        // - Fixed, time-invariant surface temperature flux mode:
+        if (pp.contains("surface_temp_flux")) {
+            m_surf_temp_flux_use_table = false;
+
+            pp.query("surface_temp_flux", m_mo.surf_temp_flux);
+
+            amrex::Print()
+                << "ABLWallFunction: Fixed, time-invariant surface temperature "
+                   "flux mode is selected."
+                << std::endl;
+        }
+        // - surface temperature flux from time-table mode:
+        else if (pp.contains("surface_temp_flux_timetable")) {
+            m_surf_temp_flux_use_table = true;
+
+            pp.query("surface_temp_flux_timetable", m_surf_temp_flux_timetable);
+
+            amrex::Print()
+                << "ABLWallFunction: Surface temperature flux time table "
+                   "mode is selected."
+                << std::endl;
+            if (!m_surf_temp_flux_timetable.empty()) {
+                std::ifstream ifh(m_surf_temp_flux_timetable, std::ios::in);
+                if (!ifh.good()) {
+                    amrex::Abort(
+                        "Cannot find surface_temp_flux_timetable file: " +
+                        m_surf_temp_flux_timetable);
+                }
+                amrex::Real data_time, data_value;
+                while (ifh >> data_time) {
+                    m_surf_temp_flux_time.push_back(data_time);
+                    m_surf_temp_flux_value.push_back(data_value);
+                    ifh.ignore(
+                        std::numeric_limits<std::streamsize>::max(), '\n');
```

Comment:
These code lines are here to skip the header line in the data file. This ignore statement needs to be first, prior to the loop.